File: bestia-models/src/main/java/net/bestia/model/battle/DamageType.java
package net.bestia.model.battle;

public enum DamageType {
	
	/**
	 * Damage is used as heal.
	 */
	HEAL, 
	
	/**
	 * Damage was missed.
	 */
	MISS,

	/**
	 * Normal hit damage.
	 */
	HIT,

	/**
	 * This was a critical damage.
	 */
	CRITICAL,

	/**
	 * True damage will (in most cases) hit the bestia without modifications
	 * of status effects or equipments.
	 */
	TRUE
}

File: bestia-models/src/main/kotlin/net/bestia/model/battle/DamageType.kt
package net.bestia.model.battle

enum class DamageType {

  /**
   * Damage is used as heal.
   */
  HEAL,

  /**
   * Damage was missed.
   */
  MISS,

  /**
   * Normal hit damage.
   */
  HIT,

  /**
   * This was a critical damage.
   */
  CRITICAL,

  /**
   * True damage will (in most cases) hit the bestia without modifications
   * of status effects or equipments.
   */
  TRUE
}